reactor-netty5-core/src/main/java/reactor/netty5/NettyOutbound.java

```diff
@@ -236,7 +237,13 @@ default NettyOutbound sendGroups(Publisher<? extends Publisher<? extends Buffer>
 				    .concatMap(p -> Flux.<Buffer>from(p)
 				                        .concatWith(Mono.just(BOUNDARY.copy(0, BOUNDARY.readableBytes(), true))), 32)
 				    .doFinally(sig -> BOUNDARY.close()),
-				ReactorNetty.PREDICATE_GROUP_FLUSH);
+				buf -> {
+					boolean flush = PREDICATE_GROUP_FLUSH.test(buf);
+					if (flush) {
+						buf.close();
+					}
+					return flush;
+				});
```

Comment:
I'm wondering whether we can do this directly in https://github.com/reactor/reactor-netty/blob/fbde6d7186a99511996f78708d491d4dee5c15a1/reactor-netty5-core/src/main/java/reactor/netty5/channel/MonoSend.java#L63-L69